runtime/src/accounts_db.rs

```diff
@@ -5966,38 +5983,80 @@ impl AccountsDb {
             })
             .sum();
 
+        // Need to add these last, otherwise older updates will be cleaned
+        for slot in slots {
+            self.accounts_index.add_root(slot, false);
+        }
+
+        let timings = self.initialize_storage_count_and_alive_bytes();
         let timings = GenerateIndexTimings {
             scan_time,
             index_time: index_time.as_us(),
             insertion_time_us: insertion_time_us.load(Ordering::Relaxed),
             min_bin_size,
             max_bin_size,
             total_items,
+            ..timings
         };
         timings.report();
+    }
 
-        // Need to add these last, otherwise older updates will be cleaned
-        for slot in slots {
-            self.accounts_index.add_root(slot, false);
+    fn calculate_storage_count_and_alive_bytes(
+        &self,
+        timings: &mut GenerateIndexTimings,
+    ) -> HashMap<usize, (usize, usize)> {
+        // look at every account in the account index and calculate for each storage: stored_size and count
+        let mut storage_size_accounts_map_time = Measure::start("storage_size_accounts_map");
+        let mut maps = self
+            .accounts_index
+            .account_maps
+            .par_iter()
+            .map(|i| {
```

Comment:
`i` we usually use for integer index, seems this is actually an account entry, something like a `Rwlock<SlotList>`.